war/src/main/java/io/lumeer/engine/controller/BatchFacade.java

```diff
@@ -154,18 +157,19 @@ private void internalExecuteBatch(final MergeBatch batch) throws DbException, In
             doc.put(batch.getResultAttribute(), subDoc);
          }
 
+         String collectionCode = collectionMetadataFacade.getCollectionCode(batch.getCollectionName());
```

Comment:
I think there should be `collectionCode` stored in `Batch` objects. But it is probably outside of the scope of this pull-request.